lib/evaluateFunctions.js

```diff
@@ -96,4 +99,389 @@ function absoluteValue(node) {
+
+      newNode = NodeCreator.operator('^', [baseNode, newExponentNode]);
+      return NodeStatus.nodeChanged(
+        MathChangeTypes.CANCEL_ROOT, node, newNode);
+    }
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Returns the nthRoot evaluated for a multiplication node.
+// Expects a multiplication node.
+function nthRootMultiplication(node) {
+  let newNode = clone(node, false);
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+
+  let substeps = [];
+  let status;
+  if (NodeType.isConstant(rootNode) && !Negative.isNegative(rootNode)) {
+    status = factorChildren(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+    }
+
+    status = groupTermsByRoot(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+    }
+
+    status = convertMultiplicationToExponent(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+      if (newNode.args[0].op === '^') {
+        status = nthRootExponent(newNode);
+        substeps.push(status);
+        return NodeStatus.nodeChanged(
+          MathChangeTypes.NTH_ROOT_VALUE, node, status.newNode, true, substeps);
+      }
+    }
+  }
+
+  status = distributeNthRoot(newNode);
+  if (status.hasChanged()) {
+    substeps.push(status);
+    newNode = NodeStatus.resetChangeGroups(status.newNode);
+  }
+
+  status = evaluateNthRootForChildren(newNode);
+  if (status.hasChanged()) {
+    substeps.push(status);
+    newNode = NodeStatus.resetChangeGroups(status.newNode);
+
+    status = combineRoots(newNode);
+    if (status.hasChanged()) {
+      substeps.push(status);
+      newNode = NodeStatus.resetChangeGroups(status.newNode);
+    }
+
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.NTH_ROOT_VALUE, node, newNode, true, substeps);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Given an nthRoot node with a constant positive root, will do the step of
+// factoring all the arguments under the radicand
+function factorChildren(node) {
+  let newNode = clone(node, false);
+  const radicandNode = node.args[0];
+
+  let children = [];
+  for (let i = 0; i < radicandNode.args.length; i++) {
+    const child = radicandNode.args[i];
+
+    if (NodeType.isConstant(child) && !Negative.isNegative(child)) {
+      const radicandValue = parseFloat(child.value);
+      const factors = Factor.getPrimeFactors(radicandValue);
+      const factorNodes = factors.map(NodeCreator.constant);
+      children = children.concat(factorNodes);
+    }
+    else {
+      children.push(child);
+    }
+  }
+
+  if (children.length !== radicandNode.args.length) {
+    newNode.args[0] = NodeCreator.operator('*', children);
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.FACTOR_INTO_PRIMES, node, newNode);
```

Comment:
Let's scrap it everywhere, as long as it doesn't change. In this case, the oldNode is always the original node since I separated each step into a different function. So as long as that's true, we should scrap it.